src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -572,11 +586,57 @@ private void ReceivePoll(Internal.Poll<K, V> poll)
             }
         }
 
+        private void CommitAggregatedOffsets()
+        {
+            if(_commitMaps.Count == 0) return;
+            var aggregatedOffsets = AggregateOffsets(_commitMaps);
+            // commits can occur after the partition has been revoked from the consumer, so ensure that we only attempt to
```

Comment:
This comment details a big part of how we do rebalance handling - we filter failed / will-fail-on-next-attempt commits out here after those partitions are no longer assigned to us.